backend/test/integration/upgrade_test.go

```diff
@@ -47,22 +46,22 @@ func TestUpgrade(t *testing.T) {
 func (s *UpgradeTests) TestPrepare() {
 	t := s.T()
 
-	test.DeleteAllJobs(s.jobClient, s.resourceNamespace, t)
 	test.DeleteAllRuns(s.runClient, s.resourceNamespace, t)
+	test.DeleteAllJobs(s.jobClient, s.resourceNamespace, t)
```

Comment:
The rationale is to delete children resources first. Run may have parent Jobs. No strong opinion here.

Reply:
Jobs aren't really parents of runs, when we delete a job, I would never imagine we delete the past runs scheduled by the job. I think we should revert to the original order to keep the test robust. Otherwise, there would be possibility of uncleaned runs mess with test result, and it would not be obvious to debug.

Reply:
Sorry, I meant "creator job" instead of parent. I moved it back in the upgrade test. If you meant to revert it everywhere, let me know.